xgcm/test/test_metric_ops_multi_axis.py

```diff
@@ -46,15 +44,30 @@ def test_bgrid(self, funcname, boundary, periodic):
             ["dx_t", "dy_t", "dz_t", "area_t", "volume_t"],
             ["xt", "yt", "zt", ["xt", "yt"], ["xt", "yt", "zt"]],
         ):
-            new = func(ds.tracer, axis, **kwargs)
+            if funcname == "cumint":
+                kwargs = dict(boundary=boundary, fill_value=fill_value)
+                new = func(ds.tracer, axis, **kwargs)
+            else:
+                new = func(ds.tracer, axis, boundary, fill_value)
```

Comment:
this could be combined to only one line with no if conditions

Reply:
I rewrote `cumint` so it expects `kwargs` input, because it passes that on to `cumsum`, which has other `kwargs` such as `keep_coords`. I rewrote `average` and `integrate` to accept `boundary` and `fill_value` so they can pass on the values to `get_metric`. These are set to `None` by default. Both `average` and `integrate` have kwargs as possible  inputs but these seem to be passed on to xarray operations rather than within `grid.py`.

Reply:
See my comment above: I believe boundary and fill_value should be explicit kwargs in all of those functions. Then you can also simplify this here.

Reply:
`kwargs` in `cumint` is passed on to `cumsum` which can accept boundary and fill_value
`kwargs` in `average` and `integrate` are passed on to `xarray.mean` and `xarray.sum`
if I just include boundary and fill_value as part of `kwargs` for average and integrate I get the following error statements:
```python
TypeError: nansum() got an unexpected keyword argument 'boundary'
TypeError: mean() got an unexpected keyword argument 'boundary'
```

Reply:
See lines 1813 and 1865

Reply:
Yes instead of `kwargs` lets add `boundary` and `fill_value` to the signature of `average` and `integrate`. That way it gets intercepted, is not passed on to xarray, and users know that they can specify it using autocomplete.

Reply:
> Yes instead of `kwargs` lets add `boundary` and `fill_value` to the signature of `average` and `integrate`. 

Is this what you mean?

```python
def average(self, da, axis, boundary, fill_value, **kwargs):
        """
        Perform weighted mean reduction along specified axis or axes,
        accounting for grid metrics. (e.g. cell length, area, volume)

        Parameters
        ----------
        axis : str, list of str
            Name of the axis on which to act
        boundary : str or dict, optional,
            boundary can either be one of {None, 'fill', 'extend', 'extrapolate'}
            * None:  Do not apply any boundary conditions. Raise an error if
              boundary conditions are required for the operation.
            * 'fill':  Set values outside the array boundary to fill_value
              (i.e. a Dirichlet boundary condition.)
            * 'extend': Set values outside the array to the nearest array
              value. (i.e. a limited form of Neumann boundary condition where
              the difference at the boundary will be zero.)
            * 'extrapolate': Set values by extrapolating linearly from the two
              points nearest to the edge
            This sets the default value. It can be overriden by specifying the
            boundary kwarg when calling specific methods.
        fill_value : float, optional
            The value to use in the boundary condition when `boundary='fill'`.
        **kwargs: dict
            Additional arguments passed to `xarray.DataArray.weighted.mean`

        Returns
        -------
        da_i : xarray.DataArray
            The averaged data
        """
```

Reply:
Then cumint will still use kwargs?

```python
def cumint(self, da, axis, **kwargs):
        """
        Perform cumulative integral along specified axis or axes,
        accounting for grid metrics. (e.g. cell length, area, volume)

        Parameters
        ----------
        axis : str, list of str
            Name of the axis on which to act
        %(grid_func.parameters)s

        Returns
        -------
        da_i : xarray.DataArray
            The cumulatively integrated data
        """
```

Reply:
Yes but with default `boundary=None, fill_value=None`. In this case, it should pull the default from `Axis.boundary` and `Axis.fill_value` automatically ( but this should be tested )